opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import argparse
+import os
+import pickle
+import random
+import shutil
+import subprocess
+import time
+from typing import List
+from zipfile import ZipFile
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+import yaml
+from accelerate import Accelerator, notebook_launcher
+from sentence_transformers import InputExample, SentenceTransformer
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import TrainingArguments, get_linear_schedule_with_warmup
+
+
+class SentenceTransformerModel:
+    def __init__(self, model_url: str = None) -> None:
+        """
+        Description:
+        Initiate a sentence transformer model object.
+
+        Parameters:
+        model_url: str = None
+             the url to download sentence transformer model, if None, default as 'sentence-transformers/msmarco-distilbert-base-tas-b'
+        Return:
+             None
+        """
+        if model_url is None:
+            self.model_url = "sentence-transformers/msmarco-distilbert-base-tas-b"
+        else:
+            self.model_url = model_url
+
+    def train(
+        self,
+        read_path: str = None,
+        output_model_name: str = None,
+        output_model_path: str = None,
+        zip_file_name: str = None,
+        use_accelerate: bool = False,
+        compute_environment: str = None,
+        num_machines: int = 1,
+        num_processes: int = None,
+        learning_rate: float = 2e-5,
+    ) -> None:
+        """
+        Description:
+        Read the synthetic queries and use it to fine tune/train (and save) a sentence transformer model.
+
+        Parameters:
+        read_path: str
+            required, path to the zipped file that contains generated queries, if None, raise exception.
```

Comment:
I think we should not set default value if some parameter is required, change `read_path: str = None,` to `read_path: str,` ?

Reply:
the rest of the py-ml package all have default values assigned for parameters, is it better to follow the same coding style? https://github.com/elastic/eland/blob/main/eland/dataframe.py#L330

Reply:
or we can match the rest of the package and only not assigned default for "required" field, what do you think?

Reply:
Agree, I think we should not set default value for "required" field.